src/aspire/utils/indices.py

```diff
@@ -0,0 +1,36 @@
+"""
+Utility functions for common index arithmetic.
+"""
+
+import numpy as np
+import random
+
+def random_tri_indices(nsamples, n, k=0, m=None,
+                       triu_or_tril='triu', seed=None):
+    """
+    Returns random nsamples of indices from a trianglular matrix.
+
+    Indices are returned as a list of 2-tuples.
+
+    :param nsamples: Number of samples to return.
+    :param n: The number of rows of the matrix.
+    :param k: Optional, diagonal offset; defaults 0.
+    :param m: Optional, number of cols of matrix; defaults=n.
+    :param triu_or_tril: Optional, `triu` or `tril` indices; defaults `triu`.
+    :param seed: Optional, specify rng seed, note resets random state.
+
+    :return: Random sample of indices as list of 2-tuples.
+    """
+
+    if triu_or_tril == 'triu':
+        ind_f = np.triu_indices
+    elif triu_or_tril == 'tril':
+        ind_f = np.tril_indices
+    else:
+        raise RuntimeError('triu_or_tril must be one of `triu` or `tril`.')
+
+    if seed is not None:
```

Comment:
Since we are using the seed here, other functions calling this function should have the option to set the seed too. We need to make the  corresponding changes later.

Reply:
The code as it is written actually does not use the seed here unless explicit overridden.  Forcing the seed is only to facilitate testing, and is just a place holder until such time that actual software engineering takes place.  I think we've talked about this several times now and there is an issue about it.